core/src/main/java/org/mapfish/print/wrapper/PJoinedArray.java

```diff
@@ -0,0 +1,118 @@
+package org.mapfish.print.wrapper;
+
+/**
+ * PArray that is consists of several PArray objects merged together.
+ *
+ * @author Jesse on 4/18/2015.
+ */
+public final class PJoinedArray implements PArray {
+    private final PArray[] arrays;
+    private final int combinedSize;
+
+    /**
+     * Constructor.
+     * @param arrays the arrays that constitute the joined array.
+     */
+    public PJoinedArray(final PArray[] arrays) {
+        this.arrays = arrays;
+        int size = 0;
+        for (PArray array : arrays) {
+            size += array.size();
+        }
+
+        this.combinedSize = size;
+    }
+
+    @Override
+    public int size() {
+        return this.combinedSize;
+    }
+
+    @Override
+    public PObject getObject(final int i) {
+        int index = i;
```

Comment:
That would require an API change of an API that goes back to V1.  I'd rather leave the API alone.  Or if we want to change it we can do it in a separate PR.